src/bin/version/main.rs

```diff
@@ -0,0 +1,101 @@
+extern crate core;
+
+use colored::Colorize;
+use std::ffi::OsString;
+use std::path::Path;
+use std::{env, fs};
+
+const SDKMAN_DIR_ENV_VAR: &str = "SDKMAN_DIR";
+const DEFAULT_SDKMAN_HOME: &str = ".sdkman";
+const VERSION_FILE: &str = "var/version";
+
+fn main() {
+    let sdkman_dir = infer_sdkman_dir();
+    let version = locate_version_file(sdkman_dir).and_then(read_content);
+    match version {
+        Some(content) => println!("\n{} {}", "SDKMAN".yellow(), content.yellow()),
+        None => std::process::exit(exitcode::CONFIG),
+    }
+}
+
+fn infer_sdkman_dir() -> String {
+    match env::var(SDKMAN_DIR_ENV_VAR) {
+        Ok(s) => s,
+        Err(_) => fallback_sdkman_dir(),
+    }
+}
+
+fn fallback_sdkman_dir() -> String {
+    dirs::home_dir()
+        .map(|dir| dir.join(DEFAULT_SDKMAN_HOME))
+        .unwrap()
```

Comment:
Just using `unwrap` here because things _should_ never go wrong here (TM) :smile:

Reply:
Good point! Will this also work reliably on Windows? I remember looking for a solution to find the HOME directory regardless of platform and it seemed that this was basically impossible to do reliably.